tests/integration/test_provider.py

```diff
@@ -133,41 +140,73 @@ async def test_remove_application_removes_user_and_acls(ops_test: OpsTest, usern
 
 
 @pytest.mark.abort_on_fail
-async def test_change_client_topic(ops_test: OpsTest):
-    action = await ops_test.model.units.get(f"{DUMMY_NAME_2}/0").run_action("change-topic")
-    await action.wait()
+async def test_deploy_producer_same_topic(ops_test: OpsTest, usernames):
+    """Test the correct deployment and relation with role producer."""
+    app_charm = await ops_test.build_charm("tests/integration/app-charm")
```

Comment:
Do we have to build the app-charm for every test? 

Could we add this to some object and refer to it every time we want to re-deploy?